odl/discr/grid.py

```diff
@@ -1264,7 +1168,7 @@ def uniform_sampling_fromintv(intv_prod, shape, nodes_on_bdry=True):
             gmin.append(xmin + (xmax - xmin) / (2 * n))
             gmax.append(xmax - (xmax - xmin) / (2 * n))
 
-    return RegularGrid(gmin, gmax, shape)
+    return uniform_grid(gmin, gmax, shape)
 
 
 def uniform_sampling(min_pt, max_pt, shape, nodes_on_bdry=True):
```

Comment:
This uses the interval product min/max, while the other uses the grid min/max directly. Subtle difference.